### PR TITLE
Merged request

### PR DESCRIPTION
**Pull request by AnushaTandon**

### CODE REVIEW

File: spec/factories/merge_requests.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :merge_request do
+
```
Comment (ror-mahesh) [inline spec/factories/merge_requests.rb:3]: if we not using then remove  it.

File: spec/factories/merged_requests.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :merged_request do
+
```
Comment (ror-mahesh) [inline spec/factories/merged_requests.rb:3]: same

File: spec/models/merge_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
```
Comment (ror-mahesh) [inline spec/models/merge_request_spec.rb:2]: same

File: spec/models/merged_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe MergedRequest, type: :model do
```
Comment (ror-mahesh) [inline spec/models/merged_request_spec.rb:3]: same

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/controllers/merged_requests_controller.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
+class MergedRequestsController < InheritedResources::Base
+  before_action :set_task
+  before_action :set_merged_request, only: :update
```
[the remark this answers is not in the document]
Comment (AnushaTandon) [reply app/controllers/merged_requests_controller.rb:3]: but i used only update for ajax calling in update action.

[the remark this answers is not in the document]
Comment (ror-mahesh) [reply app/controllers/merged_requests_controller.rb:3]: no,
We are using same query in set_merged_request, edit and destroy.
So why we not use the same filter for all of thease?

File: app/controllers/merged_requests_controller.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  def show
+    redirect_to project_task_path(@task)
+  end
+
+  private
+
+  def set_task
+    @task = Task.find_by_id(params[:task_id])
+    @project = @task.project if @task.present?
```
[the remark this answers is not in the document]
Comment (ror-mahesh) [reply app/controllers/merged_requests_controller.rb:44]: which type of error.
I think in your database there are wrong data also present..

File: app/views/merged_requests/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      <%= form.label :raised_date %>
+      <%= form.date_select :raised_date, class: 'form-control', selected: @merged_request.raised_date %>
+   </div>
+   <div class="mb-3">
+      <label for="message-text" class="col-form-label">Description:</label>
+      <%= form.text_area :description, class: 'form-control', selected: @merged_request.description %>
+   </div>
+   <div class="mb-3">
+      <%= form.label :status %>
+      <%= form.select :status, [['Open', 'open'], ['Closed', 'closed'], ['Draft', 'draft'], ['Merged', 'merged']], {}, class: 'form-control', selected: @merged_request.status %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_form.html.erb:17]: can we use status as enum.

File: app/views/merged_requests/_mr.html.erb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    <thead>
+      <tr>
+         <th scope="col">url_link</th>
+         <th scope="col">raised_date</th>
+         <th scope="col">description</th>
+         <th scope="col">status</th>
+      </tr>
+   </thead>
+   <tbody id= "merge">
+      <% @task.merged_requests&.each do |merge_request| %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_mr.html.erb:11]: is & needed?

File: app/views/merged_requests/_mr.html.erb [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            <%= merge_request.raised_date %>
+         </td>
+         <td>
+            <%= merge_request.description %>
+         </td>
+         <td>
+            <%= merge_request.status %>
+         </td>
+          <td>
+            <% if merge_request&.id&.present? %> <%= link_to 'Edit' , edit_task_merged_request_path(@task, merge_request), class: "btn btn-primary mb-3 right", data: { bs_toggle: "modal",bs_target: "#exampleModal",remote: true } %> <% end %>
```
Comment (ror-mahesh) [inline app/views/merged_requests/_mr.html.erb:26]: same.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

**Pull request merged**